README.md

```diff
@@ -12,40 +12,58 @@
-- SoC Requirement 21 wird nicht direkt umgesetzt, sondern muss auf dem Sprungserver davor, der Serviceshell, realisiert werden.
+## Module Description
 
-##Usage
-- Das Modul sollte 1-zu-1 in die Projekt-Repositories übernommen werden können.
-###Usage ohne Puppet
-- Die Templates ssh_config.erb und sshd_config.erb können, bis auf den Bereich "ListenAdress" und des "SFTP Subsystem" in der sshd_config.erb 1-zu-1 übernommen werden. Der Disclaimer mit dem Bezug auf Puppet sollte ebenfalls entfernt werden.
-###Verifikation
-- Die Verifikation des sicheren Moduls kann über Serverspec (s. Serverspec im Repo) oder Nessus Auditfile getestet werden.
+This module controls /etc/ssh/sshd_config, /etc/ssh/ssh_config, /etc/issue and /etc/motd
 
-##Reference
-- SSH-Anforderungen stammen aus [PSA 07 2015](https://www.telekom.com/psa).
+### sshd_config
 
-##Limitations
-- Modul wurde erfolgreich gegen CentOS6, RHEL6, RHEL7 getestet.
+- SoC Requirements 1 - 8, (9), 10-12 are fulfilled without restrictions.
+- SoC Requirement 9 has a parameter, because that module is used on our administration jump hosts, where we need the agent forwarding.
+  - default parameter is compliant
+- SoC Requirements 13, (14), 17, 18 are fulfilled via public key authentication and their settings.
+  - challengeresponseauthentication has a parameter, because that's sometimes necessary for bootstrapping. Some images are allowing that initially, and are hardened after provisioning.
+- SoC Requirement 15 is not compliant by default, because other modules are being used for user management, which are either not supporting the change of the key folders or do not secure that folder according to the requirement.
+  - This module atleast can be adjusted to a matching module, by adjusting the parameter AuthorizedKeysFile.
+- SoC Requirement 16 has to be taken care by every project individually.
+- SoC Requirement 21 (the daemon part) is compliant without restrictions.
 
-##Development
-- Aenderungen am Modul sollten auch im Serverspec Script secc_ssh_spec.rb nachgezogen werden.
+### ssh_config
+
+- SoC Requirements 19, 22 are fulfilled without restrictions.
+- SoC Requirement 20 (security of the private key) cannot be verified by ssh client or daemon.
+- SoC Requirement 21 (usage of the ssh auth-agent) cannot be verified by ssh client or daemon.
+
+## Usage
+
+- Either include the module via git or puppetforge. (T-Systems is using r10k)
+
+### Usage ohne Puppet
+
+- The templates, ssh_config.erb and sshd_config.erb, can be used without puppet. There is no deep ruby or puppet knowhow needed.
+- The disclaimer, that those files are managed by puppet, should be removed.
+
+### Verification
+
+- The tests (spec/acceptance) can be used to verify the hardening, even without using the module on your servers.
+
+## Reference
+
+- The hardening requirements are from Telekom Security, Privacy and Security Assessment - 3_04_Secure_Shell_(SSH). Last verified on 2018-07-31.
```

Comment:
Can you add a link to the psa here? (https://www.telekom.com/psa).

Reply:
That cannot be resolved externally?